src/lpython/parser/parser.yy

```diff
@@ -491,6 +492,8 @@ for_statement
         $$ = FOR_01($2, $4, $7, @$); }
     | KW_FOR tuple_item KW_IN expr ":" sep statements KW_ELSE ":"
         sep statements { $$ = FOR_02($2, $4, $7, $11, @$); }
+    | KW_FOR tuple_item KW_IN expr ":" TK_TYPE_COMMENT TK_NEWLINE statements {
+        $$ = FOR_03($2, $4, $6, $8, @$); }
     ;
```

Comment:
Can you parse the for-else loop with the type_comments as well? Also, add a test for it.

Reply:
Done!

Reply:
Thank you!